### PR TITLE
Rename trie/TransactionStorage to Storage (#1783)

### PR DESCRIPTION
**Pull request by nsiregar**
Rename `trie/TransactionStorage` to `Storage` to resolve issue #1783

### CODE REVIEW

File: core/trie/trie_test.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ func TestPutZero(t *testing.T) {
 		var keys []*felt.Felt
 
 		// put random 64 keys and record roots
-		for range 64 {
+		for range make([]int, 64) {
```
Comment (IronGauntlets) [inline core/trie/trie_test.go:167]: Is this change required?

File: core/trie/trie_test.go [hunk 2]
```diff
@@ -202,7 +202,7 @@ func TestPutZero(t *testing.T) {
 		t.Run("remove keys one by one, check roots", func(t *testing.T) {
 			var gotRoot *felt.Felt
 			// put zero in reverse order and check roots still match
-			for i := range 64 {
+			for i := range make([]int, 64) {
```
Comment (IronGauntlets) [inline core/trie/trie_test.go:205]: I don't think this change is required either

**Review by IronGauntlets — CHANGES_REQUESTED** [review]
Please see the comments above

File: core/trie/trie_test.go (hunk 1 above)
Comment (nsiregar) [reply core/trie/trie_test.go:167]: I added this change because `golangci-lint` raise `cannot range over 64 (untyped int constant) (typecheck)`.
should i revert this part?

File: core/trie/trie_test.go (hunk 2 above)
Comment (nsiregar) [reply core/trie/trie_test.go:205]: similar with previous, changes added because `golangci-lint` typecheck

Comment (nsiregar) [reply core/trie/trie_test.go]: reverted to previous state

File: core/trie/trie_test.go (hunk 1 above)
Comment (nsiregar) [reply core/trie/trie_test.go]: reverted to previous state

**Comment by nsiregar** [comment]
Update as requested

**Comment by derrix060** [comment]
@github-actions run

<details>
<summary>🚀 Merge preview</summary>

```js
(async () => {
  // Get pull-req URL like "https://api.github.com/repos/nwtgck/actions-merge-preview/pulls/4"
  const pullReqUrl = context.payload.issue.pull_request.url;
  const githubUser = context.payload.repository.owner.login;
  const res = await fetch(pullReqUrl, {
    headers: [
      ['Authorization', `Basic ${Buffer.from(`${githubUser}:${githubToken}`).toString('base64')}`]
    ]
  });
  const resJson = await res.json();
  const prUserName = resJson.head.user.login;
  const baseBranchName = resJson.base.ref;
  const branchName = resJson.head.ref;
  const fullRepoName = resJson.head.repo.full_name;
  const previewBranchName = `actions-merge-preview/${prUserName}-${branchName}`;
  execSync(`git config --global user.email "github-actions[bot]@users.noreply.github.com"`);
  execSync(`git config --global user.name "github-actions[bot]"`);
  // (from: https://stackoverflow.com/a/23987039/2885946)
  execSync(`git fetch --all`);
  console.log(execSync(`git checkout ${baseBranchName}`).toString());
  console.log(execSync(`git checkout -b ${previewBranchName} ${baseBranchName}`).toString());
  console.log(execSync(`git pull https://github.com/${fullRepoName}.git ${branchName}`).toString());
  // Push preview branch
  // NOTE: Force push (should be safe because preview branch always start with "actions-merge-preview/")
  execSync(`git push -fu origin ${previewBranchName}`);
  const baseRepoFullName = context.payload.repository.full_name;
  // Create GitHub client
  const githubClient = new GitHub(githubToken);
  // Comment body
  const commentBody = `🚀 Preview branch:  \n<https://github.com/${baseRepoFullName}/tree/${previewBranchName}>`;
  // Comment the deploy URL
  await postComment(commentBody);
})();
```
</details>

**Review by IronGauntlets — APPROVED** [review]

**Review by kirugan — APPROVED** [review]
